src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadataSpec.scala

```diff
@@ -148,66 +159,100 @@ class DatasetMetadataSpec extends TestSupportFixture {
-      t.getMessage shouldBe expectedMessage
-    }
+    """{ "contributors": [ { "insertions": "van der", "organization": "University of Zurich" } ] }"""
+      .causesInvalidDocumentException("""requirement failed: Author has no surname so neither titles nor insertions; got {"insertions":"van der","organization":"University of Zurich"} Author""")
+  }
+
+  "DatasetMetadata.dates" should "reject dcterms:dateSubmitted" in {
+    """{"dates": [{ "qualifier": "dcterms:dateSubmitted", "value": "2018-12", "scheme": "dcterms:W3CDTF" }]}"""
+      .causesInvalidDocumentException("""requirement failed: No dcterms:dateSubmitted allowed; got [{"scheme":"dcterms:W3CDTF","value":"2018-12","qualifier":"dcterms:dateSubmitted"}]""")
+  }
+
+  it should "reject multiple dcterms:available" in {
+    """{"dates": [
+      |  { "qualifier": "dcterms:available", "value": "2018", "scheme": "dcterms:W3CDTF" },
+      |  { "qualifier": "dcterms:available", "value": "2018-12", "scheme": "dcterms:W3CDTF" },
+      |  { "qualifier": "dcterms:created", "value": "2018-12", "scheme": "dcterms:W3CDTF" },
+      |]}""".stripMargin
+      .causesInvalidDocumentException("""requirement failed: At most one allowed; got [{"scheme":"dcterms:W3CDTF","value":"2018","qualifier":"dcterms:available"},{"scheme":"dcterms:W3CDTF","value":"2018-12","qualifier":"dcterms:available"}]""")
+  }
+
+  "DatasetMetadata.audience" should "reject an audience without a scheme" in {
+    """{"audiences": [{ "key": "yyy", "value": "" }]}"""
+      .causesInvalidDocumentException("""don't recognize {"audiences":[{"key":"yyy","value":""}]}""")
+  }
+
+  it should "reject an audience with an empty string for a schema" in {
+    """{"audiences": [{ "scheme": "", "key": "xxx", "value": "yyy" }]}"""
+      .causesInvalidDocumentException("""requirement failed: Empty string for a mandatory field; got {"scheme":"","key":"xxx","value":"yyy"} SchemedKeyValue""")
+  }
+
+  it should "reject an audience with an empty string for a key" in {
+    """{"audiences": [{ "scheme": "xxx", "key": "", "value": "yyy" }]}"""
+      .causesInvalidDocumentException("""requirement failed: Empty string for a mandatory field; got {"scheme":"xxx","key":"","value":"yyy"} SchemedKeyValue""")
+  }
+
+  it should "reject an audience with an empty string for a value" in {
+    """{"audiences": [{ "scheme": "xxx", "key": "yyy", "value": "" }]}"""
+      .causesInvalidDocumentException("""requirement failed: Empty string for a mandatory field; got {"scheme":"xxx","key":"yyy","value":""} SchemedKeyValue""")
+  }
+
+  "DatasetMetadata.spatialBoxes" should "reject a non-numeric value" in {
+    """{"spatialBoxes": [ { "scheme": "RD", "north": "486890,5", "east": 121811.88, "south": 436172.5,  "west": 91232.016 }]}""".stripMargin
+      .causesInvalidDocumentException("""requirement failed: Invalid number [486890,5]; got {"scheme":"RD","north":"486890,5","east":"121811.88","south":"436172.5","west":"91232.016"} SpatialBox""")
```

Comment:
@rvanheest @lindareijnhoudt 
Note that the UI can but does not need to quote the numbers!

Reply:
But it will get them quoted from the server.

Reply:
@jo-pol thanks for the heads up!

> Note that the UI can but does not need to quote the numbers!

As of https://github.com/DANS-KNAW/easy-deposit-ui/pull/84 we changed this to align with this PR: it will serve strings to the client, rather than numbers

> But it will get them quoted from the server.

[That was already in place from the beginning.](https://github.com/DANS-KNAW/easy-deposit-ui/blob/c37d0ffedc26c1ea05d4693c68d95b9b743462ca/src/main/typescript/lib/metadata/SpatialBox.ts#L36-L39)